lib/PuppeteerSharp/Input/Keyboard.cs

```diff
@@ -109,8 +103,11 @@ public async Task TypeAsync(string text, TypeOptions options = null)
             {
                 delay = (int)options.Delay;
             }
-            foreach (var letter in text)
+
+            var textParts = StringInfo.GetTextElementEnumerator(text);
```

Comment:
why was this change required?

Reply:
I was just preparing a post about this. Emojis (let's call it Unicode chars) uses 4 bytes. So you need 2 chars to represent one Unicode char. If you do `string.ToCharArray` and send one by one to Chromium you would get invalid characters (a.k.a. □) on the other side.
So you need a method, `GetTextElementEnumerator`, that returns the "elements" of the text. `GetTextElementEnumerator` is able to return a \u1F479 (which are 2 chars: \u1F47 and 9) as one element.
From the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.stringinfo.gettextelementenumerator?view=netframework-4.7.2):
>The .NET Framework defines a text element as a unit of text that is displayed as a single character, that is, a grapheme. A text element can be a base character, a surrogate pair, or a combining character sequence.